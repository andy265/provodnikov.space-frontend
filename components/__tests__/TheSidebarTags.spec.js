const Vuex = require('vuex')
const { createRenderer } = require('vue-server-renderer')
const { createLocalVue, shallow } = require('vue-test-utils')
const TheSidebarTags = require('~/components/TheSidebarTags')

describe('TheSidebarTags.vue', () => {
  let localVue = createLocalVue()
  localVue.use(Vuex)

  let store = new Vuex.Store({
    state: {
      tags: {
        list: [
          { id: 'tag_1', name: 'tag name 1' },
          { id: 'tag_2', name: 'tag name 2' }
        ]
      }
    }
  })

  store.state.tags.list.toString = () => JSON.stringify(store.state.tags.list)

  it('has same HTML structure', () => {
    let Component = Object.assign({}, TheSidebarTags, { components: {} })
    let wrapper = shallow(Component, {
      localVue,
      store
    })
    let renderer = createRenderer()
    // eslint-disable-next-line
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })
})
