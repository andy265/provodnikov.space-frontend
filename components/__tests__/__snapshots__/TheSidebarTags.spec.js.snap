// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheSidebarTags.vue has same HTML structure 1`] = `
<div class="the-sidebar-tags">
    <AppTags tags="[{&quot;id&quot;:&quot;tag_1&quot;,&quot;name&quot;:&quot;tag name 1&quot;},{&quot;id&quot;:&quot;tag_2&quot;,&quot;name&quot;:&quot;tag name 2&quot;}]" class="app-tags_direction_column"></AppTags>
</div>
`;
